client/state/data-layer/wpcom/me/purchases/schema.js

```diff
@@ -0,0 +1,75 @@
+/** @format */
+
+/**
+ * This schema is necessarily fairly relaxed.
```

Comment:
Do you think it's fair to say that such a non-strict schema doesn't help too much?

Reply:
The goal is that no "bad" data gets into Calypso.

As @dmsnell mentions, we're free to throw errors in the fromApi transformer, which will put us into the same flow as schema validation errors.

It seems the schema is most important for things that won't fail in transformation:

- Required properties
- expected-to-be-well-typed properties

Anything that we don't expect to be well typed coming from the may be best handled in the transformer where we can ensure data is present and of the correct type.

At the moment, this PR doesn't include a transformer, however I plan to split this up into smaller PRs so it may make sense to include it during that process.